File: client/crypto/simple_crypt.cpp
#include "simple_crypt.h"

QByteArray SimpleCrypt::encode(QByteArray data, int key)
{
    for (int i = 0; i < data.length(); i++)
        data[i] = data[i] + key;
    
    return data;
}

QByteArray SimpleCrypt::decode(QByteArray data, int key)
{
    for (int i = 0; i < data.length(); i++)
        data[i] = data[i] - key;
    
    return data;
}

File: server/src/main/java/amalgama/chat/network/crypt/SimpleCrypt.java
package amalgama.chat.network.crypt;

public class SimpleCrypt {
    public static String encode(String data, int key) {
        byte[] bytes = data.getBytes();
        for (int i = 0; i < bytes.length; i++) {
            bytes[i] += key;
        }
        return new String(bytes);
    }

    public static String decode(String data, int key) {
        byte[] bytes = data.getBytes();
        for (int i = 0; i < bytes.length; i++) {
            bytes[i] -= key;
        }
        return new String(bytes);
    }
}
